pkg/agent/datamodel/sig_config.go

```diff
@@ -0,0 +1,417 @@
+package datamodel
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	AzurePublicCloudSigTenantID     string = "33e01921-4d64-4f8c-a055-5bdaffd5e33d" // AME Tenant
+	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
+)
+
+//SIGAzureEnvironmentSpecConfig is the overall configuration differences in different cloud environments.
+// TODO(tonyxu) merge this with AzureEnvironmentSpecConfig from aks-engine(pkg/api/azenvtypes.go) once it's moved into AKS RP
+type SIGAzureEnvironmentSpecConfig struct {
+	CloudName                string                    `json:"cloudName,omitempty"`
+	SigTenantID              string                    `json:"sigTenantID,omitempty"`
+	SubscriptionID           string                    `json:"subscriptionID,omitempty"`
+	SigUbuntuImageConfig     map[Distro]SigImageConfig `json:"sigUbuntuImageConfig,omitempty"`
+	SigCBLMarinerImageConfig map[Distro]SigImageConfig `json:"sigCBLMarinerImageConfig,omitempty"`
+	SigWindowsImageConfig    map[Distro]SigImageConfig `json:"sigWindowsImageConfig,omitempty"`
+	//TODO(adadilli) add PIR constants as well
+}
+
+// SIGConfig is used to hold configuration parameters to access AKS VHDs stored in a SIG
+type SIGConfig struct {
+	TenantID       string                      `json:"tenantID"`
+	SubscriptionID string                      `json:"subscriptionID"`
+	Galleries      map[string]SIGGalleryConfig `json:"galleries"`
+}
+
+type SIGGalleryConfig struct {
+	GalleryName   string `json:"galleryName"`
+	ResourceGroup string `json:"resourceGroup"`
+}
+
+type SigImageConfigOpt func(*SigImageConfig)
+
+// GetCloudTargetEnv determines and returns whether the region is a sovereign cloud which
+// have their own data compliance regulations (China/Germany/USGov) or standard
+// Azure public cloud or air gapped clouds
+// NOTE(tonyxu)
```

Comment:
should we remove some of these comments here

Reply:
Yeah, sure. I removed those comments.